source/tools/CustomHostCpuManager.h

```diff
@@ -0,0 +1,477 @@
+            } else {
+                unboostBandwidth();
+            }
+
+            HostTools::sleepUntilNanoTime(wakeupTime_ns);
+        }
+
+        /* Read starting time of the callback */
+        getDeadlineAttribute(mDLStart);
+        mStartingTime_ns = HostTools::getNanoTime();
+    }
+
+    /**
+     * This is called when the application determines its workload for the
+     * callback.
+     *
+     * It calculates the runtime parameter to be assigned to the SCHED_DEADLINE
+     * Linux scheduling class, which adapts the CPU frequency accordingly.
+     *
+     * @param currentWorkUnits the current work units requested by the
+     * application, representing the computational effort that is going to be
+     * requested in the future computations. The meaning of this value may be
+     * different for different applications, e.g. it can be the number of notes
+     * played or the number of filters applied to a given signal.
+     * @param maxWorkUnits (not used) the maximum number of work units the
+     * application is planning to request.
+     */
+    void setApplicationLoad(int32_t currentWorkUnits,
+                            int32_t maxWorkUnits) override {
+        /*
+         * Perform an update only when the number of work units is changed, or
+         * if the mBWUpdateCounter is 0 (periodic update).
+         */
+        if (currentWorkUnits != getCurrentWorkUnits() ||
+                maxWorkUnits != getMaxWorkUnits() ||
+                mBWUpdateCounter == 0) {
+            int64_t expectedRuntime_ns;
+            double bandwidth;
+
+            /*
+             * mBWUpdateCounter is reset, to prevent a close bandwith updates
+             * due to the timer, after a workUnit change. This allows the
+             * statistics for the new workUnits to stabilize.
+             */
+            if (mBWUpdateCounter != 0)
+                mBWUpdateCounter = 0;
+
+            expectedRuntime_ns = mCpuTiming.getWorkUnitRuntime(currentWorkUnits);
+
+            /*
+             * When we read a 0 runtime for != 0 workUnits, it means that
+             * something bad is going on, so, set the maximum runtime.
+             * This may happen when the linear regression fails.
+             */
+            if (expectedRuntime_ns == 0 && currentWorkUnits != 0)
+                bandwidth = mMaxBW;
+            else
+                bandwidth = static_cast<double>(expectedRuntime_ns) / mPeriod_ns;
+
+            /* Add some offset to the computed bandwidth */
+            bandwidth *= 1.1;
+            bandwidth += 0.05;
+
+            /* Bound the bandwidth to the limit set by the Kernel */
+            if (bandwidth > mMaxBW)
+                bandwidth = mMaxBW;
+
+            if (currentWorkUnits != getCurrentWorkUnits() ||
+                    maxWorkUnits != getMaxWorkUnits()) {
+                /* Print the current status only when the workUnits changed */
+                printf("workUnits=%d\trequestDLBandwidth=%f",
```

Comment:
The print could disrupt the benchmark. May need to use Log.

Reply:
OK, I think this is something that can be done in the future.